src/transformers/models/trillsson_efficientnet/feature_extraction_trillsson_efficientnet.py

```diff
@@ -0,0 +1,513 @@
+        self.f_min = f_min
+        self.mel_high_frequency_q = mel_high_frequency_q
+        self.mel_break_frequency_hertz = mel_break_frequency_hertz
+        self.fft_length = fft_length
+
+    @classmethod
+    def pad_symmetric(cls, raw_speech: np.ndarray, padding: int):
+        """Symmetric pad a 2D Tensor"""
+        if raw_speech.ndim != 2:
+            raise ValueError(f"Raw speech must be rank 2 got {raw_speech.shape}")
+        raw_speech_len = raw_speech.shape[1]
+        if padding > raw_speech_len:
+            pad_raw_speech = cls.repeat_n_times_with_extra(raw_speech, padding, raw_speech_len)
+        else:
+            pad_raw_speech = np.pad(raw_speech, [(0, 0), (0, padding)], mode="symmetric")
+        return pad_raw_speech
+
+    @classmethod
+    def pad_waveform(cls, raw_speech: np.ndarray, padding: int, mode: str, padding_value: float):
+        """Pad waveform"""
+        if raw_speech.ndim != 2:
+            raise ValueError(f"Raw speech must be rank 2 got {raw_speech.shape}")
+        if mode == "symmetric":
+            return cls.pad_symmetric(raw_speech, padding)
+        else:
+            pad_raw_speech = np.pad(raw_speech, [(0, 0), (0, padding)], mode=mode, constant_values=padding_value)
+            return pad_raw_speech
+
+    @staticmethod
+    def repeat_n_times_with_extra(raw_speech: np.ndarray, padding: int, raw_speech_len: int):
+        """Pad symmetric longer than the original raw speech"""
+        if raw_speech.ndim != 2:
+            raise ValueError(f"Raw speech must be rank 2 got {raw_speech.shape}")
+        num_copies = np.floor_divide(padding, raw_speech_len)
+        r = np.fliplr(raw_speech)
+        f = np.concatenate((r, raw_speech), axis=1)
+        copies = np.tile(f, [1, np.floor_divide(num_copies, 2)])
+        if num_copies % 2 != 0:
+            copies = np.concatenate((copies, r), axis=1)
+        pre_pad = np.concatenate([raw_speech, copies], axis=1)
+        extra = padding % raw_speech_len
+        pad_raw_speech = np.pad(pre_pad, [(0, 0), (0, extra)], mode="symmetric")
+        return pad_raw_speech
+
+    @staticmethod
+    def stabilized_log(data: np.ndarray, floor: float, additive_offset: float):
+        """Stabilized MelSpectrogram"""
+        return np.log(np.maximum(data, floor) + additive_offset)
+
+    @staticmethod
+    def rescale(input_array: np.ndarray, rate: float = 1.0 / 128.0, offset: int = -1):
+        """Rescale the input array"""
+        return input_array * rate + offset
+
+    @staticmethod
+    def hann_window(window_length, periodic=True, a=0.5, b=0.5, dtype=np.float32):
+        """Generate a [Hann window][hann]."""
+
+        if window_length == 1:
+            return np.ones([1], dtype=dtype)
+
+        even = 1 - np.mod(window_length, 2)
+        n = window_length + periodic * even - 1
+        count = np.arange(window_length, dtype=dtype)
+        cos_arg_np = 2 * np.pi * count / n
+
+        return (a - b * np.cos(cos_arg_np)).astype(dtype)
```

Comment:
There is also `np.hanning(window_length)` , but I am guessing you need the `a` and `b` to be variable? If not, let's remove this

Reply:
I will remove variables a and b. The reason I have to reimplement the Hann window is that the outputs from tf.signal.hann_window and np.hanning are different